File: android/app/src/main/java/com/awaytime/app/service/FoxMoodService.kt
package com.awaytime.app.service

import android.content.Context
import android.content.SharedPreferences
import kotlinx.coroutines.flow.MutableStateFlow
import kotlinx.coroutines.flow.StateFlow
import kotlinx.coroutines.flow.asStateFlow

class FoxMoodService(private val context: Context) {
    private val prefs: SharedPreferences = context.getSharedPreferences("fox_mood_prefs", Context.MODE_PRIVATE)
    
    private val _currentMood = MutableStateFlow(FoxMood.HAPPY)
    val currentMood: StateFlow<FoxMood> = _currentMood.asStateFlow()
    
    private val _usagePercent = MutableStateFlow(0f)
    val usagePercent: StateFlow<Float> = _usagePercent.asStateFlow()
    
    init {
        // Load saved mood
        val savedMood = prefs.getString("current_mood", FoxMood.HAPPY.name)
        _currentMood.value = FoxMood.valueOf(savedMood ?: FoxMood.HAPPY.name)
    }
    
    fun updateUsagePercent(percent: Float) {
        _usagePercent.value = percent
        val newMood = determineMoodFromUsage(percent)
        
        if (newMood != _currentMood.value) {
            _currentMood.value = newMood
            saveMood(newMood)
        }
    }
    
    private fun determineMoodFromUsage(percent: Float): FoxMood {
        return when {
            percent <= 40f -> FoxMood.HAPPY
            percent <= 70f -> FoxMood.SLEEPY
            percent <= 90f -> FoxMood.SAD
            else -> FoxMood.EXHAUSTED
        }
    }
    
    private fun saveMood(mood: FoxMood) {
        prefs.edit().putString("current_mood", mood.name).apply()
    }
    
    fun getMoodMessage(mood: FoxMood): String {
        return when (mood) {
            FoxMood.HAPPY -> "Great job! You're managing your screen time well."
            FoxMood.SLEEPY -> "You're doing okay, but consider taking a break soon."
            FoxMood.SAD -> "You're getting close to your limit. Time for a break?"
            FoxMood.EXHAUSTED -> "You've reached your limit. Please take a break!"
        }
    }
    
    fun getMoodColor(mood: FoxMood): Long {
        return when (mood) {
            FoxMood.HAPPY -> 0xFF4CAF50 // Green
            FoxMood.SLEEPY -> 0xFFFF9800 // Orange
            FoxMood.SAD -> 0xFF2196F3 // Blue
            FoxMood.EXHAUSTED -> 0xFF9E9E9E // Gray
        }
    }
}

File: ios/Awaytime/Services/FoxMoodService.swift
import Foundation
import SwiftUI

class FoxMoodService: ObservableObject {
    @Published var currentMood: FoxMood = .happy
    @Published var usagePercent: Float = 0.0
    
    private let userDefaults = UserDefaults.standard
    private let currentMoodKey = "current_fox_mood"
    
    init() {
        loadSavedMood()
    }
    
    func updateUsagePercent(_ percent: Float) {
        usagePercent = percent
        let newMood = FoxMood.fromUsagePercent(percent)
        
        if newMood != currentMood {
            currentMood = newMood
            saveMood(newMood)
        }
    }
    
    private func loadSavedMood() {
        if let savedMoodString = userDefaults.string(forKey: currentMoodKey),
           let savedMood = FoxMood(rawValue: savedMoodString) {
            currentMood = savedMood
        }
    }
    
    private func saveMood(_ mood: FoxMood) {
        userDefaults.set(mood.rawValue, forKey: currentMoodKey)
    }
    
    func getMoodMessage(for mood: FoxMood) -> String {
        switch mood {
        case .happy:
            return "Great job! You're managing your screen time well."
        case .sleepy:
            return "You're doing okay, but consider taking a break soon."
        case .sad:
            return "You're getting close to your limit. Time for a break?"
        case .exhausted:
            return "You've reached your limit. Please take a break!"
        }
    }
    
    func getMoodColor(for mood: FoxMood) -> Color {
        switch mood {
        case .happy:
            return .green
        case .sleepy:
            return .orange
        case .sad:
            return .blue
        case .exhausted:
            return .gray
        }
    }
}